src/Analyzers/Core/Analyzers/AbstractBuiltInCodeStyleDiagnosticAnalyzer_Core.cs

```diff
@@ -25,6 +27,14 @@ internal abstract partial class AbstractBuiltInCodeStyleDiagnosticAnalyzer : Dia
             bool isUnnecessary,
             bool configurable)
         {
+            // 'isUnnecessary' should be true only for sub-types of AbstractBuiltInUnnecessaryCodeStyleDiagnosticAnalyzer.
+            // NOTE: AbstractParenthesesDiagnosticAnalyzer is an exception as it is a common sub-type for
+            // AbstractRemoveUnnecessaryParenthesesDiagnosticAnalyzer (unnecessary code analyzer) and
+            // AbstractAddRequiredParenthesesDiagnosticAnalyzer (non-unnecessary code analyzer).
```

Comment:
i'm fine with you breaking the inheritance in AbstractParenthesesDiagnosticAnalyzer , it's not really helpful at all afaict.  it literally just contains a few helpers that could be in a static class.